src/brogue/Rogue.h

```diff
@@ -1024,7 +1024,7 @@ enum tileFlags {
     IS_IN_MACHINE               = (IS_IN_ROOM_MACHINE | IS_IN_AREA_MACHINE),    // sacred ground; don't generate items here, or teleport randomly to it
 
     PERMANENT_TILE_FLAGS = (DISCOVERED | MAGIC_MAPPED | ITEM_DETECTED | HAS_ITEM | HAS_DORMANT_MONSTER
-                            | HAS_STAIRS | SEARCHED_FROM_HERE | PRESSURE_PLATE_DEPRESSED
+                            | HAS_MONSTER | HAS_STAIRS | SEARCHED_FROM_HERE | PRESSURE_PLATE_DEPRESSED
```

Comment:
Is there any possibility of allies blocking each other from the stairs, as this flag is not updated as off-level pathing happens?

Reply:
Off-level pathing is determined from this map in startLevel: `calculateDistances(mapToStairs, px, py, (flying ? T_OBSTRUCTS_PASSABILITY : T_PATHING_BLOCKER) | T_SACRED, NULL, true, true);`
It simply does not look at HAS_MONSTER during off-level pathing, but pushes the monster to a location without one later in restoreMonster.